armory/metrics/statistical.py

```diff
@@ -21,6 +22,54 @@ def register(metric):
     return set_namespace(registered, metric)
 
 
+@populationwise
+def precision_and_recall(y, y_pred):
+    """
+    Produce a dictionary whose keys are class labels, and values are (precision, recall) for that class
+    """
+    # Assumes that every class is represented in y
+
+    C = confusion_matrix(y, y_pred, normalize=False)
+    # breakpoint()
+    N = C.shape[0]
+    D = {}
+    for class_ in range(N):
+        # precision: true positives / number of items identified as class_
+        tp = C[class_, class_]
+        total_selected = C[:, class_].sum()
+        precision = tp / total_selected
+
+        # recall: true positives / number of actual items in class_
```

Comment:
per-class recall is the exact same as per-class accuracy, which I didn't realize till now.  Is it still useful to have two separate `per_class_accuracy` and `per_class_precision_and_recall` functions?

Reply:
Sure, let's keep both.